lib/trivial.nix

```diff
@@ -205,6 +205,13 @@ rec {
   */
   inNixShell = builtins.getEnv "IN_NIX_SHELL" != "";
 
+  /* Determine whether the function is being called from inside pure-eval mode
+     by seeing whether `builtins` contains `currentSystem`. If not, we must be in
+     pure-eval mode.
+
+     Type: inPureEvalMode :: bool
+  */
+  inPureEvalMode = ! builtins ? currentSystem;
```

Comment:
While I still think this function to be useful, maybe it is better to introduce this in another PR since it is not used by anything right now :thinking: ?

Reply:
I've done that here. https://github.com/NixOS/nixpkgs/pull/157480, I have also rebased this PR to remove this commit, as well as removing all of the conflation in the initial commit.